src/epd_driver/highlevel.c

```diff
@@ -59,7 +59,7 @@ enum EpdDrawError epd_hl_update_screen(EpdiyHighlevelState* state, enum EpdDrawM
 
 EpdRect _inverse_rotated_area(uint16_t x, uint16_t y, uint16_t w, uint16_t h) {
   // If partial update uses full screen do not rotate anything
-  if (x != 0 && y != 0 && EPD_WIDTH != w && EPD_HEIGHT != h) {
+  if (!(x == 0 && y == 0 && epd_rotated_width() == w && epd_rotated_height() == h)) {
```

Comment:
This is not correct. This function do not exist and the name is **epd_rotated_display_width()**
It's also not been tested since running the demo the Jiraffe in portrait mode it comes cut at half. 
IMHO This is the correct line:

    if (!(x == 0 && y == 0 && EPD_WIDTH == w && EPD_HEIGHT == h)) {
    
 This has to check width like this, not calling the functions that retrieve the "rotated dimensions"

Reply:
I can't follow you. Using `epd_rotated_display_width()` and `epd_rotated_display_height()` respectively, works perfectly fine. I couldn't spot any difference in the demo.

If you remember, could you please provide a screenshot? Maybe it has to do with the individual framework and board configuration.